documentation/src/main/asciidoc/user_guide/clustering.adoc

```diff
@@ -1,211 +1,308 @@
-
-But why would one use a local cache rather than a map? Caches offer a lot of features
-over and above a simple map, including write-through and write-behind caching to persist
-data, eviction of entries to prevent running out of memory, and support for expirable
-entries.
-Infinispan, specifically, is built around a high-performance, read-biased data container
-which uses modern techniques similar to
-link:http://en.wikipedia.org/wiki/Read-copy-update[read-copy-update]
--- which buys you non-blocking, thread-safe reads even when concurrent writes are taking
-place.
-Infinispan also makes heavy use of compare-and-swap and other lock-free algorithms,
-making it ideal for high-throughput, multi-CPU/multi-core environments.
-Further, Infinispan's Cache API extends the JDK's ConcurrentMap - making migration from a
+In this mode, it acts as a simple, in-memory data cache similar to the old JBoss Cache
+and EHCache.
+
+But why would one use a local cache rather than a map?
+Caches offer a lot of features over and above a simple map, including write-through and
+write-behind to a persistent store, eviction of entries to prevent running out of memory,
+and expiration.
+
+Infinispan's `Cache` interface extends JDK's `ConcurrentMap` -- making migration from a
 map to Infinispan trivial.
 
-Creating a local-only cache-manager is trivial: just use the _no-argument_ DefaultCacheManager
-constructor, or supply the following XML configuration file
+Infinispan caches also support transactions, either integrating with an existing
+transaction manager or running a separate one.
+Local caches transactions have two choices:
 
-[source,xml]
-----
-<infinispan/>
+1. When to lock?
+*Pessimistic locking* locks keys on a write operation or when the user calls
+`AdvancedCache.lock(keys)` explicitly.
+*Optimistic locking* only locks keys during the transaction commit, and instead it throws
+a `WriteSkewCheckException` at commit time, if another transaction modified the same keys
+after the current transaction read them.
+
+2. Isolation level.
+We support *read-committed* and *repeatable read*.
 
-----
-You can also create local caches in a clustering-capable CacheManager.
 
 include::simple_cache.adoc[]
 
+
+=== Invalidation Mode
+In invalidation, the caches on different nodes do not actually share any data.
+Instead, when a key is written to, the cache only aims to remove data that may be stale
+from other nodes.
+This cache mode only makes sense if you have another, permanent store for your data such
+as a database and are only using Infinispan as an optimization in a read-heavy system,
+to prevent hitting the database for every read.
+If a cache is configured for invalidation, every time data is changed in a cache, other
+caches in the cluster receive a message informing them that their data is now stale and
+should be removed from memory and from any local store.
+
+.Invalidation mode
```

Comment:
I've already seen a question about clearing invalidation cache... May I ask you to say a few words how to do it?